lib/validator/task/mod.rs

```diff
@@ -1155,15 +1155,446 @@ where
         }
     };
 
+    let deserialization_duration = start.elapsed() - fetch_duration;
+
+    tracing::debug!(
+        "Fetched ({:?}) and deserialized ({:?}) batch of {} block(s) with {} total transactions",
+        fetch_duration,
+        deserialization_duration,
+        blocks.len(),
+        blocks.iter().map(|block| block.txdata.len()).sum::<usize>(),
+    );
+
     Ok(blocks)
 }
 
+fn handle_block_batch(
+    dbs: &Dbs,
+    blocks: &[Block],
+    event_tx: &Sender<Event>,
+) -> Result<(), error::Sync> {
+    let start = Instant::now();
+
+    let mut total_txs = 0;
+    let mut total_bmm_commitments = 0;
+    let mut total_events = 0;
+
+    // Do a single DB transaction for the entire batch. DB commits are a big part
+    // of the sync time, so we want to reduce the number of times we do this.
+    let mut rwtxn = dbs.write_txn()?;
+
+    // Process blocks sequentially to maintain ordering and database consistency
+    for block in blocks {
+        let block_hash = block.block_hash();
+
+        tracing::trace!("Syncing block #{} `{block_hash}`", {
+            // Do the data fetch within the macro, to avoid the cost on higher
+            // log levels
+            dbs.block_hashes.height().get(&rwtxn, &block_hash)?
+        });
+
+        let start_block = Instant::now();
+        // We should not call out to `invalidateblock` in case of failures here,
+        // as that is handled by the cusf-enforcer-mempool crate.
+        // FIXME: handle disconnects
+        let event = connect_block(&mut rwtxn, dbs, block)?;
+
+        let connect_block_duration =
+            jiff::SignedDuration::try_from(start_block.elapsed()).unwrap_or_default();
+
+        // Create dynamic fields using a HashMap for structured logging
+        match &event {
+            Event::ConnectBlock {
+                header_info,
+                block_info,
+            } => {
+                // Keep all the blocks at info level in the beginning,
+                // and then taper off into less log noise
+                let modulo = match header_info.height {
```

Comment:
Can you give this a better name than `modulo`? Maybe `log_interval`?

Also this could be implemented much more succinctly as
```rust
let exp = (n / 100).checked_ilog10().unwrap_or(0);
10.pow(exp)
```